### PR TITLE
docs: reorganize and refresh readme

### LINKED ISSUE #1734: Documentation Updates

### PR DESCRIPTION
**Pull request by sskeirik**
Fixes: #1734 

I wrote this after walking through the K Framework build process with a client and noticed that there were some dependencies that were listed in the "short version" but not the "long version."

I updated the "long version" section to move the list of the dependencies to the top and then put the installation notes at the bottom. I think this reads a bit more nicely --- it gives you an upfront picture of the entire process without getting bogged in the details at first.

EDIT: I also made a few other changes:

- I added a quick description of what the K Framework is.
- I added a table of contents
- I added some notes about supported system configurations
- I reorganized the section on building the OCaml backend slightly
- I added some notes about building the K Framework on macOS/brew systems
- I renamed and reorganized the "Installing on a Fresh Windows Subsystem for Linux"

Questions:

- Do we officially support 32 bit systems? It's not clear to me.
- Is the readme getting to be too long and verbose? Should we break it out into separate documents?

Thoughts?

### CODE REVIEW

**Comment by sskeirik** [comment]
@ehildenb @dwightguth @ttuegel This PR is for updating the main README for the reasons stated in the original description. I'm not sure who should review this.

That said, do you guys have info about my pending questions above?
Any feedback in general would be appreciated.

File: README.md
Comment (ehildenb) [inline README.md]: Likely some of this should go into `k-distribution/INSTALL.md` instead, and you should link there from here. Anything to do with getting K standing up on a given platform from our packaged releases should go into `INSTALL.md`. This guide should focus on (i) from-source builds for developers, and (ii) how developers can get examples running so they can begin working on K.

**Comment by ehildenb** [comment]
@sskeirik it looks mostly OK so far. I would emphasize that packaged installation instructions should go into `k-distribution/INSTALL.md`, and that the `README.md` is intended _for developers_ to be able to begin working on K, and doesn't make it into any user-facing generated documentation.

**Comment by sskeirik** [comment]
@ehildenb I updated the README as suggested and moved the quick start guide section into the INSTALL.md file.

The quick start could be further parameterized so that it applies to multiple systems --- let me know if you are interested in that.

Besides that one possible issue, I think this is ready for re-review.

File: README.md
Comment (ehildenb) [inline README.md]: This is packaging information and should remain in k-distribution/INSTALL.md. We do not want to duplicate information whenever possible, rather link to the one true source.

File: README.md
Comment (ehildenb) [inline README.md]: We do not need to include instructions in the K README about setting up virtual development environments. Users can google that on their own. You can include a single sentence or two saying that they will need to set it up, but instruct them to look up their own guides.

File: k-distribution/INSTALL.md
Comment (ehildenb) [inline k-distribution/INSTALL.md]:
```suggestion
Quick Start Guide
-----------------
```

**Comment by ehildenb** [comment]
Why is there a haskell backend update in this PR? Please revert that.

**Comment by ehildenb** [comment]
Please also remove `k-distribution/README.md`, and change the `src/main/assembly` XML file to reflect that (use the top-level README).

**Comment by ehildenb** [comment]
We do _not_ support 32-bit machines.

**Comment by sskeirik** [comment]
@ehildenb @dwightguth The requested changes have been made and this should be ready for re-review.

Note that README.md file currently links to the INSTALL.md file on the master branch (so it will be out of date if you click on it), but will work correctly once this PR is merged. This is helpful because relative URLs break during the package assembly process (unless we do some sort of URL rewriting which seems overkill).

**Review by ehildenb — APPROVED** [review]

**Pull request merged**
